If you call a function which can return a null pointer you probably want
to check the return value before attempting to dereference it. </sarcasm>

--- a/sys-utils/lscpu.c
+++ b/sys-utils/lscpu.c
@@ -966,7 +966,7 @@
 		*(p - 2) = '\0';
 		add_summary_s(tb, sec, _("CPU op-mode(s):"), buf);
 	}
-	if (ct->addrsz)
+	if (ct && ct->addrsz)
 		add_summary_s(tb, sec, _("Address sizes:"), ct->addrsz);
 #if !defined(WORDS_BIGENDIAN)
 	add_summary_s(tb, sec, _("Byte Order:"), "Little Endian");
@@ -1008,9 +1008,9 @@
 	sec = NULL;
 
 	/* Section: cpu type description */
-	if (ct->vendor)
+	if (ct && ct->vendor)
 		sec = add_summary_s(tb, NULL, _("Vendor ID:"), ct->vendor);
-	if (ct->bios_vendor)
+	if (ct && ct->bios_vendor)
 		add_summary_s(tb, sec, _("BIOS Vendor ID:"), ct->bios_vendor);
 
 	for (i = 0; i < cxt->ncputypes; i++)
